src/Google/GlobalSiteTag.php

```diff
@@ -156,14 +195,241 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
+			esc_js( $product->get_id() ),
+			esc_js( (string) $product->get_price() ),
+			esc_js( $product->get_name() ),
+			esc_js( join( '& ', $product->get_categories() ) ),
+		);
+	}
+
+	/**
+	 * Display the JavaScript code to track all pages.
+	 */
+	public function display_page_view_event_snippet(): void {
+		printf(
+			'<script>
+                gtag(
+                    "event", "page_view", {
+                    "send_to": "GLA",
+                    "developer_id.%s": "true",});
+			</script>',
+			esc_js( self::DEVELOPER_ID )
+		);
+	}
+
+	/**
+	 * Display the JavaScript code to track the cart page.
+	 */
+	public function display_cart_page_snippet(): void {
+		// Only display on the cart page.
+		if ( ! is_cart() ) {
+			return;
+		}
+
+		$item_info = '';
+
+		foreach ( WC()->cart->get_cart() as $cart_item ) {
+			// gets the product id
+			$id = $cart_item['product_id'];
+
+			// gets the product object
+			$product = $cart_item['data'];
+			$name    = $product->get_name();
+			$price   = $product->get_price();
+			// gets the cart item quantity
+			$quantity = $cart_item['quantity'];
+
+			$item_info = $item_info . sprintf(
+				'{
+				"id": "gla_%s",
+				"price": %s,
+				"google_business_vertical": "retail",
+				"name":"%s",
+				"quanitity":"%s",
+				}',
+				esc_js( $id ),
+				esc_js( $price ),
+				esc_js( $name ),
+				esc_js( $quantity )
+			);
+		}
+		$value = WC()->cart->total;
+		printf(
+			'<script>
+            gtag("event", "page_view",
+				{"send_to": "GLA",
+				"ecomm_pagetype": "cart",
+				"value": "%s",
+				 items: [' . esc_html( $item_info ) . ']}); </script>',
+			esc_js( $value ),
+		);
+	}
+
+	/**
+	 * Display the JavaScript code to track the purchase page.
+	 */
+	public function display_purchase_page_snippet(): void {
+		// Only display on the order confirmation page.
+		if ( ! is_order_received_page() ) {
+			return;
+		}
+		$order_id = $this->wp->get_query_vars( 'order-received', 0 );
+		if ( empty( $order_id ) ) {
+			return;
+		}
+		$order = wc_get_order( $order_id );
+
+		$item_info = '';
+		foreach ( $order->get_items() as $item_id => $item ) {
+			$product_id   = $item->get_product_id();
+			$product_name = $item->get_name();
+			$quantity     = $item->get_quantity();
+			$price        = $item->get_subtotal();
+			$item_info    = $item_info . sprintf(
+				'{
+                    "id": "gla_%s",
+                    "price": %s,
+                    "google_business_vertical": "retail",
+                    "name":"%s",
+                    "quanitity":"%s",
+                }',
+				esc_js( $product_id ),
+				esc_js( $price ),
+				esc_js( $product_name ),
+				esc_js( $quantity ),
+			);
+
+		}
+
+		$is_new_customer = false;
+		if ( $order->get_user_id() ) {
+			$total_orders = wc_get_customer_order_count( $order->get_user_id() );
+		} else {
+			$total_orders = WC_Order_Export_Data_Extractor::get_customer_order_count_by_email( $order->get_billing_email() );
```

Comment:
This is not a function that is available in WooCommerce, it seems to be from the plugin `woo-order-export-lite`.

Probably better to use `wc_get_orders` here setting the [customer](https://github.com/woocommerce/woocommerce/wiki/wc_get_orders-and-WC_Order_Query#customer) and making sure to return only ids (otherwise it will load all the order objects).

I think we should extract this to a separate function which returns a boolean `$is_new_customer` so we can clearly define what we are doing within the function.

Reply:
Great idea! We don't need all the data information of the order. We only want to count the number of the previous orders to check if this is first time customer. Returning ids should be enough.